File: Java/485. Max Consecutive Ones.java
/**
 * https://leetcode.com/problems/max-consecutive-ones/
 */

public class Solution {
    // O(n) time complexity
    // O(1) space complexity
    public int findMaxConsecutiveOnes(int[] nums) {
        int res = 0;
        int count = 0;
        for (int n : nums) {
            if (n == 1)
                count++;
            else {
                res = Math.max(count, res);
                count = 0;
            }
        }
        return Math.max(count, res);
    }
}

File: Kotlin/485. Max Consecutive Ones.kt
/**
 * https://leetcode.com/problems/max-consecutive-ones/
 */

class Solution {
    // O(n) time complexity
    // O(1) space complexity
    fun findMaxConsecutiveOnes(nums: IntArray): Int {
        var res = 0
        var count = 0
        nums.forEach { n ->
            if (n == 1)
                count++
            else {
                res = Math.max(res, count)
                count = 0
            }
        }
        return Math.max(res, count)
    }
}